query/command/command.go

```diff
@@ -210,26 +221,24 @@ func (cmd *SelectCommand) Execute(context ExecutionContext) (CommandResult, erro
 			chosenTimerange.Slots(), slotLimit)
 	}
 	hasTimeout := context.Timeout != 0
-	var cancellable api.Cancellable
+	var timeoutOwner tasks.TimeoutOwner
```

Comment:
It looks like there were some bugs associated with the timeout type, actually.

A zero-value `Timeout` will give you a nil channel back, which just blocks forever if you wait on it (and a zero-value `TimeoutOwner` just contains a zero-value `Timeout`).